rust/worker/src/blockstore/arrow/provider.rs

```diff
@@ -418,10 +421,9 @@ impl SparseIndexManager {
         new_id: Uuid,
     ) -> SparseIndex {
         // TODO: error handling
-        println!("Forking sparse index from {:?}", old_id);
+        tracing::info!("Forking sparse index from {:?}", old_id);
         let original = self.get::<K::ReadableKey<'key>>(old_id).await.unwrap();
         let forked = original.fork(new_id);
-        self.cache.write().insert(new_id, forked.clone());
```

Comment:
is this safe to remove? what if someone tries to access the forked id? I think the management logic here is a bit odd to me